drivers/net/netvsc/hn_ethdev.c

```diff
@@ -1129,12 +1098,146 @@ hn_detach(struct hn_data *hv)
 	hn_rndis_detach(hv);
 }
 
+static int
+hn_dev_mtu_set(struct rte_eth_dev *dev, uint16_t mtu)
+{
+	struct hn_data *hv = dev->data->dev_private;
+	struct hn_rx_queue **rxqs = (struct hn_rx_queue **)dev->data->rx_queues;
+	struct hn_tx_queue **txqs = (struct hn_tx_queue **)dev->data->tx_queues;
+	struct rte_eth_dev *vf_dev;
+	unsigned int orig_mtu = dev->data->mtu;
+	uint32_t rndis_mtu;
+	int ret = 0;
+	int i;
+
+	if (dev->data->dev_started) {
+		PMD_DRV_LOG(ERR, "Device must be stopped before changing MTU");
+		return -EIO;
+	}
+
+	/* Change MTU of underlying VF netdev first, if it exists */
+	rte_rwlock_read_lock(&hv->vf_lock);
+	vf_dev = hn_get_vf_dev(hv);
+	if (hv->vf_ctx.vf_vsc_switched && vf_dev) {
+		ret = vf_dev->dev_ops->mtu_set(vf_dev, mtu);
```

Comment:
A VF device can be hot added or removed. Look at hn_vf_add() on setting the mtu if VF is hot added

Reply:
Added code to set the VF MTU on a hot add.